packages/opentelemetry-exporter-collector/src/transform.ts

```diff
@@ -246,8 +268,58 @@ export function groupSpansByResourceAndLibrary(
   }, new Map<Resource, Map<core.InstrumentationLibrary, ReadableSpan[]>>());
 }
 
+/**
+ * Takes an array of metrics and groups them by resource and instrumentation
+ * library
+ * @param metrics metrics
+ */
+export function groupMetricsByResourceAndLibrary(
+  metrics: MetricRecord[]
+): Map<Resource, Map<core.InstrumentationLibrary, MetricRecord[]>> {
+  return metrics.reduce((metricMap, metric) => {
```

Comment:
Use of `reduce` here feels unnecessarily fancy. Would be simpler to just make a `Map` and do a `for .. of` loop.

Reply:
Ah I see, I wrote it like this to stay consistent with similar logic elsewhere in the codebase (e.g. `groupSpanByResourceAndLibrary`)